### PR TITLE
extended existing gtest to cover a wide variety of javascript syntax …

### PR DESCRIPTION
**Pull request by justrandyc**
35 additional cases of JavaScript and expected Json conversion strings, to be fed into the JsonUtility.  Examples only of JavaScript Objects, but including arrays and subobjects.  Boolean, Numeric, and String instances in various combinations.

This gtest can be run with the other gtests as JsonUtils_unittest and was tested on Cent OS 7.
It is an expansion of the fix for issue #622

### CODE REVIEW

File: earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp [hunk 1]
```diff
@@ -23,14 +23,343 @@ namespace {
...
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// Ok so apparently using single quotes is somewhat common
+// Surely the conversion utility shouldn't just "accept" them
+// and put double quotes INSIDE of them!
+//TEST_F(JsonUnitTest, singleQuotes) {
+//    const std::string js = R"js_({'name':'fred'})js_";
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp]: Yes that is a good catch

File: earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp [hunk 2]
```diff
@@ -23,14 +23,343 @@ namespace {
...
+    const std::string expectedJson = R"js_({"fred's":"name"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// While unlikely and dangerous, this seems legal in Javascript
+// Perhaps a bogus test/unlikely edge case
+//TEST_F(JsonUnitTest, doubleQuoteInKey) {
+//    const std::string js = R"js_({"freds\"car\"":"christine"})js_";
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp]: I think you are correct here.  This is legal JS.

File: earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp [hunk 3]
```diff
@@ -23,14 +23,343 @@ namespace {
...
+    const std::string js = R"js_({"car":"fred's"})js_";
+    const std::string expectedJson = R"js_({"car":"fred's"})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// Can we escape quotes?  This is legit for Javascript
+//TEST_F(JsonUnitTest, doubleQuoteInValue) {
+//    const std::string js = R"js_({"carname":"\"christine\""})js_";
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp]: Yes this is a good test.

File: earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp [hunk 4]
```diff
@@ -23,14 +23,343 @@ namespace {
...
+}
+
+// Should it eat white space? I would think the conversion would collapse it.
+// This may not be a bug.  Yes the test is ugly on purpose to contrast output
+//TEST_F(JsonUnitTest, spansMultipleLines) {
+//    const std::string js = R"js_({name:"fred", 
+//         car:"christine"
+//        ,color:"red"})js_";
+//    const std::string expectedJson = 
+//        R"js_({"name":"fred","car":"christine","color":"red"})js_";
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp]: Yea I wasn't clear on that.  I think currently it preserves white space.  Maybe we can add an option to create compact JSON.  @tst-jhaight can confirm behavior.

File: earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp [hunk 5]
```diff
@@ -23,14 +23,343 @@ namespace {
...
+//    std::string result = JsonUtils::JsToJson(js);
+//
+//    EXPECT_EQ(expectedJson, result);
+//}
+
+// Return/NewLine should be... eaten? Replaced with a space?
+// Surely not just have extra escape chars thrown at it
+//TEST_F(JsonUnitTest, newlineInValue) {
+//    const std::string js = R"js_({"name":"fred\r\nis\r\ncool\r\n"})js_";
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp]: In this case you are stating you want new lines in your data.  This is perfectly allowed and should be transformed to JSON...  I'm wondering with the raw string you could just add literal new lines.  But in any case it should be preserved in the JSON too.

Curious what this would do:

```
R"js_({"name":"fred
is
cool
"})js_"
```

It should produce the same string for JSON (with the literal new lines).  I'm not sure how to do return carriages in a raw string as all characters are taken literally.

File: earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp [hunk 6]
```diff
@@ -23,14 +23,343 @@ namespace {
...
+    const std::string js = 
+        R"js_({names:{first:fred,new:false,house:10}})js_";
+    const std::string expectedJson = 
+        R"js_({"names":{"first":"fred","new":false,"house":10}})js_";
+    std::string result = JsonUtils::JsToJson(js);
+
+    EXPECT_EQ(expectedJson, result);
+}
+
+// possible additional tests
```
Comment (tst-rwildes) [inline earth_enterprise/src/fusion/autoingest/JsonUtils_unittest.cpp]: Yes multi-line tests would be good adds!
